data/core/units/dwarves/Miner.cfg

```diff
@@ -27,21 +27,20 @@
         icon=attacks/pick-axe.png
     [/attack]
     [standing_anim]
+        start_time=0
         [if]
             [filter]
-                role=loaded
-                [or]
-                    role=has_coal
-                [/or]
-                [or]
-                    role=has_gold
-                [/or]
+                status=has_coal,has_gold
```

Comment:
This seems to only support two types of resource, and will need some workaround if a scenario has miners picking up crystal (#7378). That workaround might be as simple as "we use has_coal to mean crystal", but is there a nicer way to incorporate it into the unit?

Reply:
I was thinking about this too a little while ago. As it is right now I don't think there's a reason to call the status "has_xxx" instead of "loaded" or similar; however, I believe I chose this with the thought that the two statuses might have different graphics at some point in the future. But I'm not sure if there's a nice way to handle this generically in a way that would support other resources…

Reply:
maybe add a generalized "has_fullbag"?

Reply:
That falls under this:

> "loaded" or similar

On the other hand, adding it as an exclusive to the existing two seems kind of pointless, as it doesn't actually help someone who wants them to be loaded with a different resource if the sprites actually reflect the resource being carried.